component/config.go

```diff
@@ -151,6 +154,13 @@ func MustNewType(strType string) Type {
 	return ty
 }
 
+func (t Type) Validate() error {
```

Comment:
@mx-psi do I remember wrongly or you were working to change this to be a struct? So this is not needed since we cannot construct an invalid type.

Reply:
Yes, see #9472 which is ready to merge. I think that PR would make this one unnecessary since it would be impossible to construct an invalid type

Reply:
> Well, technically we still [don't allow empty string names](https://github.com/open-telemetry/opentelemetry-collector/blob/94ae7762c9458cfe199cfaa3d16a4ecc1f7330be/component/config.go#L126) for type right? Given the regex does not permit empty strings

I don't think we should allow empty names, at least not through an empty `component.Type`, but I am not sure what you want to accomplish exactly, so it's better if we discuss in an issue

> (also we should probably add a generous length limit imho)

Would you be open to filing an issue about this?